### PR TITLE
Updates for ITKv5.2 code cleanups ... ROUND1

### PR DESCRIPTION
**Pull request by hjmjohnson**

### CODE REVIEW

**Comment by dzenanz** [comment]
`d:\a\1\s\include\itkEigenToMeasureImageFilter.h(96): error C3064: 'u_int8_t': must be a simple type or resolve to one` looks like it needs fixing.

**Comment by hjmjohnson** [comment]
> `d:\a\1\s\include\itkEigenToMeasureImageFilter.h(96): error C3064: 'u_int8_t': must be a simple type or resolve to one` looks like it needs fixing.

Thanks.  Fixed u_int8_t   -> uint8_t

File: include/itkEigenToMeasureImageFilter.h [hunk 1]
```diff
@@ -85,19 +85,27 @@ class ITK_TEMPLATE_EXPORT EigenToMeasureImageFilter : public ImageToImageFilter<
...
   {
     OrderByValue = 1,
     OrderByMagnitude,
     DoNotOrder
-  } EigenValueOrderType;
-  virtual EigenValueOrderType
+  };
+/**Exposes enums values for backwards compatibility*/
+#if !defined(ITK_LEGACY_REMOVE)
+  static constexpr EigenValueOrderEnum OrderByValue = EigenValueOrderEnum::OrderByValue;
```
Comment (dzenanz) [inline include/itkEigenToMeasureImageFilter.h:103]:
```suggestion
  using EigenValueOrderType = EigenValueOrderEnum;
  static constexpr EigenValueOrderEnum OrderByValue = EigenValueOrderEnum::OrderByValue;
```

File: include/itkKrcahEigenToMeasureParameterEstimationFilter.h [hunk 2]
```diff
@@ -115,13 +115,24 @@ class KrcahEigenToMeasureParameterEstimationFilter
...
     UseImplementationParameters = 1,
     UseJournalParameters
-  } KrcahImplementationType;
-  itkSetMacro(ParameterSet, KrcahImplementationType);
-  itkGetConstMacro(ParameterSet, KrcahImplementationType);
+  };
+
+  /**Exposes enums values for backwards compatibility*/
+#if !defined(ITK_LEGACY_REMOVE)
+  static constexpr KrcahImplementationEnum UseImplementationParameters =
```
Comment (dzenanz) [inline include/itkKrcahEigenToMeasureParameterEstimationFilter.h:129]:
```suggestion
  using KrcahImplementationType = KrcahImplementationEnum;
  static constexpr KrcahImplementationEnum UseImplementationParameters =
```

**Review by dzenanz — COMMENTED** [review]
We might need extra backwards compatibility definitions.

Also, it would be good if you squashed the first three commits with the last, and made it the new first commit.

File: include/itkEigenToMeasureImageFilter.h (hunk 1 above)
Comment (hjmjohnson) [reply include/itkEigenToMeasureImageFilter.h:103]: Are there packages that build on top of this?  It is not a core toolset, so I don't see the need for this extra legacy support.

**Comment by dzenanz** [comment]
d:\a\1\s\include\itkMultiScaleHessianEnhancementImageFilter.h(129): error C2146: syntax error: missing ';' before identifier 'EigenValueOrderEnum'
```
/ITKBoneEnhancement/include/itkEigenToMeasureImageFilter.h:88: error: "\ingroup BoneEnhancement" not set in class EigenValueOrderEnum.
/ITKBoneEnhancement/include/itkKrcahEigenToMeasureParameterEstimationFilter.h:118: error: "\ingroup BoneEnhancement" not set in class KrcahImplementationEnum.
/ITKBoneEnhancement/include/itkMultiScaleHessianEnhancementImageFilter.h:144: error: "\ingroup BoneEnhancement" not set in class SigmaStepMethodEnum.
10 headers checked.
```
```
/ITKBoneEnhancement/include/itkDescoteauxEigenToMeasureImageFilter.h:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkDescoteauxEigenToMeasureImageFilter.hxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkDescoteauxEigenToMeasureParameterEstimationFilter.h:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkDescoteauxEigenToMeasureParameterEstimationFilter.hxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkEigenToMeasureImageFilter.h:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkEigenToMeasureImageFilter.h:113: error: Too many semicolons
/ITKBoneEnhancement/include/itkEigenToMeasureImageFilter.hxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkEigenToMeasureParameterEstimationFilter.h:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkEigenToMeasureParameterEstimationFilter.hxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkHessianGaussianImageFilter.h:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkHessianGaussianImageFilter.hxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkKrcahEigenToMeasureImageFilter.h:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkKrcahEigenToMeasureImageFilter.hxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkKrcahEigenToMeasureParameterEstimationFilter.h:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkKrcahEigenToMeasureParameterEstimationFilter.hxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkKrcahPreprocessingImageToImageFilter.h:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkKrcahPreprocessingImageToImageFilter.hxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkMaximumAbsoluteValueImageFilter.h:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkMaximumAbsoluteValueImageFilter.h:123: error: Too many semicolons
/ITKBoneEnhancement/include/itkMultiScaleHessianEnhancementImageFilter.h:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/include/itkMultiScaleHessianEnhancementImageFilter.hxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkDescoteauxEigenToMeasureImageFilterUnitTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkDescoteauxEigenToMeasureParameterEstimationFilterUnitTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkDescoteauxEigenToScalarFunctorImageTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkDescoteauxEigenToScalarFunctorTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkDescoteauxEigenToScalarImageFilterTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkDescoteauxEigenToScalarParameterEstimationImageFilterTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkHessianGaussianImageFilterUnitTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkKrcahEigenToMeasureParameterEstimationFilterUnitTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkKrcahEigenToScalarFunctorImageFilterTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkKrcahEigenToScalarFunctorTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkKrcahEigenToScalarImageFilterTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkKrcahEigenToScalarParameterEstimationImageFilterTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkKrcahEigenToScalarPreprocessingImageToImageFilterTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkMaximumAbsoluteValueImageFilterUnitTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkMultiScaleHessianEnhancementImageFilterStaticMethodsTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
/ITKBoneEnhancement/test/itkMultiScaleHessianEnhancementImageFilterStaticMethodsUnitTest.cxx:3: error: Header mismatch: NumFOCUS (Insight) : Utilities/KWStyle/ITKHeader.h
```
```
[----------] 1 test from itkDescoteauxEigenToMeasureImageFilterUnitTest/0, where TypeParam = double
[ RUN      ] itkDescoteauxEigenToMeasureImageFilterUnitTest/0.TestDarkRealEigenPixelBrightSheet
/ITKBoneEnhancement/test/itkDescoteauxEigenToMeasureImageFilterUnitTest.cxx:249: Failure
Expected: this->m_Filter->Update() doesn't throw an exception.
```
include/itkMultiScaleHessianEnhancementImageFilter.h:129:53: error: no type named 'EigenValueOrderEnum' in namespace 'itk'

**Review by dzenanz — APPROVED** [review]

**Comment by dzenanz** [comment]
`include/itkMultiScaleHessianEnhancementImageFilter.h:129:53: warning: declaration 'using EigenValueOrderEnum = enum class itk::SymmetricEigenAnalysisEnums::EigenValueOrder' does not declare anything`

`test/itkDescoteauxEigenToMeasureParameterEstimationFilterUnitTest.cxx:149:1: warning: 'TypedTestCaseIsDeprecated' is deprecated: TYPED_TEST_CASE is deprecated, please use TYPED_TEST_SUITE [-Wdeprecated-declarations]`

**Pull request merged**